serialization/src/extras/non_empty_vec.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization_core::{Decode, Encode};
+
+/// A wrapper that saves the extra byte coming from Option by assuming the Vec can never be empty
+///
+/// Description:
+/// - If the Vec has data, it encodes to just the Vec, the Option is omitted
+/// - If the Vec has no data, it encodes to None
+/// - Some(vec![]) and None are equivalent when encoded, but when decoded result in None
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct DataOrNoVec<T>(pub Option<Vec<T>>);
```

Comment:
Is the field supposed to be public?

Reply:
Yep. How else would you access the data in it? This is just a wrapper for encoding purposes.

Reply:
Personally I prefer a method for that or an implementation of `Into` or `AsRef` traits, but it isn't that important.

Reply:
That's fair. Maybe I should do that. It's just quick and dirty, to be honest, to solve an urgent problem in the NFTs PR.